x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/DenseVector.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.vectors.query;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public interface DenseVector extends Iterable<Float> {
+    float getMagnitude();
+
+    // The implementations of these vector operations
+    double dotProduct(QueryVector queryVector);
+
+    double l1Norm(QueryVector queryVector);
+
+    double l2Norm(QueryVector queryVector);
```

Comment:
Should any of these take an additional boolean parameter to normalize the QueryVector? Or should we document that these expect the QueryVector to be normalized?

Reply:
Those don't need to be normalized, is my understanding.